tests/align/test_align_simd_base.h

```diff
@@ -0,0 +1,265 @@
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Rene Rahn <rene.rahn@fu-berlin.de>
+// ==========================================================================
+
+#ifndef TESTS_ALIGN_TEST_ALIGN_SIMD_BASE_H_
+#define TESTS_ALIGN_TEST_ALIGN_SIMD_BASE_H_
+
+#include <tuple>
+
+#include <seqan/basic.h>
+#include <seqan/align.h>
+
+#include "test_mock.h"
+
+// ----------------------------------------------------------------------------
+// Class SimdAlignTest
+// ----------------------------------------------------------------------------
+
+// Common test class instance, which stores the types to be accessed.
+template <typename TTuple>
+class SimdAlignTest : public seqan::Test
+{
+public:
+    using TAlignConfig = std::tuple_element_t<0, TTuple>;
+    using TLengthParam = std::tuple_element_t<1, TTuple>;
+    using TBandSwitch = std::tuple_element_t<2, TTuple>;
+};
+
+namespace impl
+{
+namespace test_align_simd
+{
+
+struct LocalAlignTester_
+{
+    template <typename TAlign,
+              typename TScoreValue, typename TScoreSpec,
+              typename TConfig>
+    static auto
+    run(TAlign & align,
+        seqan::Score<TScoreValue, TScoreSpec> const & score,
+        TConfig const &,
+        int const lDiag,
+        int const uDiag)
+    {
+        if (lDiag == std::numeric_limits<int>::min() && uDiag == std::numeric_limits<int>::max())
+            return localAlignment(align, score);
+        else
+            return localAlignment(align, score, lDiag, uDiag);
+    }
+};
```

Comment:
What's up with these constructs? Why did you not just use free functions? :thinking:

Reply:
Why do you bother? Second, it was already like that and I just moved it to another location so that your desktop pc can manage to build the nighties 👀

Reply:
I will merge it, but I do not approve :point_up: